Source/Schema.NET/ValuesConverter.cs

```diff
@@ -40,8 +34,147 @@ public class ValuesConverter : JsonConverter
+                    if (token.Any(t => !string.IsNullOrEmpty(GetTypeNameFromToken(t))))
+                    {
+                        var list = Activator.CreateInstance(listType);
+                        foreach (var childToken in token.Children())
+                        {
+                            var typeName = GetTypeNameFromToken(childToken);
+                            var builtType = Type.GetType($"Schema.NET.{typeName}");
+                            var child = (Thing)childToken.ToObject(builtType);
+                            listType.GetRuntimeMethod("Add", new[] { type }).Invoke(list, new object[] { child });
+                        }
+
+                        argument = list;
+                    }
+                    else
+                    {
+                        argument = token.ToObject(listType);
+                    }
+                }
+                else if (IsPrimitiveType(type))
+                {
+                    argument = value;
+                }
+                else if (type == typeof(decimal))
+                {
+                    argument = Convert.ToDecimal(value);
+                }
+                else
+                {
+                    if (type.GetTypeInfo().IsEnum)
+                    {
+                        var enumString = token.ToString().Substring("http://schema.org/".Length);
+                        argument = Enum.Parse(type, enumString);
+                    }
+                    else
+                    {
+                        var typeName = GetTypeNameFromToken(token);
+                        if (string.IsNullOrEmpty(typeName))
+                        {
+                            argument = token.ToObject(type);
+                        }
+                        else
+                        {
+                            var builtType = Type.GetType($"Schema.NET.{typeName}");
+                            argument = token.ToObject(builtType);
+                        }
+                    }
+                }
+            }
+            else
+            {
+                foreach (var type in mainType.GenericTypeArguments)
+                {
+                    try
+                    {
+                        object args;
+                        if (tokenType == JsonToken.StartObject)
+                        {
+                            var typeName = GetTypeNameFromToken(token);
+                            if (string.IsNullOrEmpty(typeName))
+                            {
+                                args = token.ToObject(type);
+                            }
+                            else if (typeName == type.Name)
+                            {
+                                args = token.ToObject(type);
+                            }
+                            else
+                            {
+                                var builtType = Type.GetType($"Schema.NET.{typeName}");
```

Comment:
This handles the case where the T is a base class, like when T is Place but the actual type is Country